### ISSUE TITLE
Improve stability checker to better detect flaky tests with timeouts

### ISSUE THREAD

**Issue by zcorpan** [w3c/web-platform-tests#10580]
From IRC
<blockquote>

&lt;jgraham> zcorpan: setTimeout is probably causing flake

&lt;jgraham> Or at least I've seen at least two recent cases where a timeout was responsible for totally unnecessary flake

&lt;jgraham> (both setTimeout rather than step_timeout, and step_timeout with a too-short timeout)

&lt;jgraham> (but if you find tests that flake and have any kind of timeout, that's the first thing to try)

&lt;jgraham> It would be ideal if we could have more stringent checks for tests with timeouts

&lt;zcorpan> jgraham: yeah, not sure how to detect it though. hmmm. for step_timeout, how about having the stability checker try running the test with a different timeout multiplier (maybe both shorter and longer, but at least longer), to see if that affects test result?

&lt;jgraham> zcorpan: I think the best thing would be to run at least one debug build in stability checks since they are more prone to flake. And maybe have more iterations required for tests that use timeouts

</blockquote>

**Comment by jugglinmike** [comment]
> I think the best thing would be to run at least one debug build in stability checks since they are more prone to flake.

@jgraham Are debug builds of Firefox publicly available? Or is this something that would have to take place within the Gecko infrastructure?

> And maybe have more iterations required for tests that use timeouts

I may be able to give this a shot later this week

**Comment by jgraham** [comment]
Debug builds are public, but there aren't convenient CDN links afaik; you  have to get them directly from CI; mozdownload supports this.

I wouldn't increase the iterations in any tets until we sort the problem where we do too many iterations and tests time out. It would also be intersting to see if running Firefox with `--verify` is more effective at finding issues because it enables chaos mode that increases the chance of races.

**Comment by zcorpan** [comment]
> I wouldn't increase the iterations in any tets until we sort the problem where we do too many iterations and tests time out.

Is that #7660 or a separate issue? If the latter, is there an issue filed already?

**Comment by jgraham** [comment]
Yes, that's #7660 although strictly it doesn't have to be "lots" of tests; 5 affected tests taking 1 minute each would exceed the 49½ minute timeout we get on Travis.
